### PR TITLE
refactor(react-grid): give column properties greater precedence

### PR DESCRIPTION
**Pull request by kvet**
BREAKING CHANGE:

A column's `getCellValue` function has gotten higher priority than that defined in the Grid component.

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/grid-core/computeds.test.js [hunk 1]
```diff
@@ -35,21 +35,19 @@ describe('GridCore Plugin computeds', () => {
...
-      expect(cellValueGetter(getCellValue, columns)({ a: 1 }, 'a'))
+      expect(cellValueGetter(undefined, columns)({ a: 1 }, 'a'))
         .toEqual(1);
-      expect(cellValueGetter(getCellValue, columns)({ b: 2 }, 'b'))
+      expect(cellValueGetter(undefined, columns)({ b: 2 }, 'b'))
         .toEqual(2);
     });
 
     it('should work with column defined cell access', () => {
       const columns = [{ name: 'a', getCellValue: row => row.b }, { name: 'b' }];
```
Comment (SergeyAlexeev) [inline packages/dx-grid-core/src/plugins/grid-core/computeds.test.js]: Column name is `a` but we get value by `b`. It's a bit confusing.

File: packages/dx-grid-core/src/plugins/grid-core/computeds.test.js [hunk 2]
```diff
@@ -35,21 +35,19 @@ describe('GridCore Plugin computeds', () => {
...
-      const getCellValue = null;
 
-      expect(cellValueGetter(getCellValue, columns)({ b: 1 }, 'a'))
+      expect(cellValueGetter(undefined, columns)({ b: 1 }, 'a'))
         .toEqual(1);
-      expect(cellValueGetter(getCellValue, columns)({ b: 2 }, 'b'))
+      expect(cellValueGetter(undefined, columns)({ b: 2 }, 'b'))
         .toEqual(2);
     });
```
Comment (SergeyAlexeev) [inline packages/dx-grid-core/src/plugins/grid-core/computeds.test.js]: I think we should add a test case that checks that `column.getCellValue` has greater priority than `Grid.getCellValue` one.

```js
const columns = [{ name: 'a', getCellValue: row => row.a }, { name: 'b' }];
const getCellValue = (row, columnName) => row[`_${columnName}`];

expect(cellValueGetter(getCellValue, columns)({ a: 1, _a: 2 }, 'a'))
   .toEqual(1);
```

File: packages/dx-grid-core/src/plugins/grid-core/computeds.test.js (hunk 1 above)
Comment (gsobolev) [reply packages/dx-grid-core/src/plugins/grid-core/computeds.test.js]: We should not write confusing tests anyway.

File: packages/dx-react-grid/docs/guides/data-accessors.md [hunk 3]
```diff
@@ -74,8 +74,6 @@ The following demo shows this approach in action:
 
 .embedded-demo(data-accessors/custom-data-accessors)
 
-Note that the Grid's `getCellValue` property has a higher priority than the column's property.
```
Comment (gsobolev) [inline packages/dx-react-grid/docs/guides/data-accessors.md]: We have changed the priorities. Why do we remove the note completely instead of changing it?

Comment (kvet) [reply packages/dx-react-grid/docs/guides/data-accessors.md]: We haven't such note in other places. We don't need to leave it here for now.

**Pull request merged**